src/syntax/translation/resugar/tests.rs

```diff
@@ -0,0 +1,383 @@
+            Box::new(concrete::Term::Universe(span(), None)),
+            Box::new(concrete::Term::Universe(span(), None)),
+        );
+
+        assert_eq!(core_term.resugar(&ResugarEnv::new()), concrete_term);
+    }
+
+    #[test]
+    fn universe0() {
+        let core_term = core::Term::universe(0);
+        let concrete_term = concrete::Term::Universe(span(), None);
+
+        assert_eq!(core_term.resugar(&ResugarEnv::new()), concrete_term);
+    }
+
+    #[test]
+    fn universe1() {
+        let core_term = core::Term::universe(1);
+        let concrete_term = concrete::Term::Universe(span(), Some(1));
+
+        assert_eq!(core_term.resugar(&ResugarEnv::new()), concrete_term);
+    }
+
+    // TODO: core::Term::Literal
+
+    #[test]
+    fn lit_bool_true() {
+        let core_term = core::Term::Literal(core::Literal::Bool(true));
+        let concrete_term = concrete::Term::Name(index(), "true".to_owned());
+
+        assert_eq!(core_term.resugar(&ResugarEnv::new()), concrete_term);
+    }
+
+    #[test]
+    fn lit_bool_false() {
+        let core_term = core::Term::Literal(core::Literal::Bool(false));
+        let concrete_term = concrete::Term::Name(index(), "false".to_owned());
+
+        assert_eq!(core_term.resugar(&ResugarEnv::new()), concrete_term);
+    }
+
+    #[test]
+    fn lit_string() {
+        let core_term = core::Term::Literal(core::Literal::String("hello".to_owned()));
+        let concrete_term =
+            concrete::Term::Literal(concrete::Literal::String(span(), "hello".to_owned()));
+
+        assert_eq!(core_term.resugar(&ResugarEnv::new()), concrete_term);
+    }
+
+    #[test]
+    fn var() {
+        let free_var = FreeVar::fresh_named("x");
+        let mut env = ResugarEnv::new();
+        env.on_item(&Label("x".to_owned()), &Binder(free_var.clone()));
+
+        let core_term = core::Term::Var(Var::Free(free_var));
+        let concrete_term = concrete::Term::Name(index(), "x".to_owned());
+
+        assert_eq!(core_term.resugar(&env), concrete_term);
+    }
+
+    #[test]
+    fn var_shadow_keyword() {
+        let free_var = FreeVar::fresh_named("if");
+        let mut env = ResugarEnv::new();
+        env.on_item(&Label("if".to_owned()), &Binder(free_var.clone()));
+
+        let core_term = core::Term::Var(Var::Free(free_var));
+        let concrete_term = concrete::Term::Name(index(), "if1".to_owned());
+
+        assert_eq!(core_term.resugar(&env), concrete_term);
+    }
+
+    #[test]
+    fn extern_() {
+        let core_term = core::Term::Extern(
+            "type".to_owned(),
+            core::RcTerm::from(core::Term::universe(0)),
+        );
+
+        let concrete_term = concrete::Term::Extern(
+            span(),
+            span(),
+            "type".to_owned(),
+            Box::new(concrete::Term::Universe(span(), None)),
+        );
+
+        assert_eq!(core_term.resugar(&ResugarEnv::new()), concrete_term);
+    }
+
+    // TODO: core::Term::Pi
```

Comment:
Gotta come back and make some tests for these, but I'm gonna call it quits on this PR for now!